src/components/panel/panel.ts

```diff
@@ -0,0 +1,30 @@
+import { BaseComponent, BaseElement, BaseOptions } from '../base';
+import { FormioComponents } from '../components';
+import { FormioTemplate } from '../../formio.template';
+import { FormArray, FormGroup, FormControl } from '@angular/forms';
+
+export interface PanelOptions extends BaseOptions<any> {
+    title?: string,
+    theme?: string,
+    components: Array<any>
+}
+
+export class PanelComponent extends BaseComponent<PanelOptions> {
+    getControl(): FormArray | FormGroup | FormControl {
+        if (!this.control) {
+            this.control = new FormGroup({});
+        }
+        return this.control;
+    }
+}
+
+export class PanelElement extends BaseElement<PanelComponent> {
+    get numComponents() : number { return this.component.settings.components.length; }
```

Comment:
Because we are now using ```formio-components```, the rendering is delegated to that library which you see by passing along the ```(render)="onRender()"``` to that component. Because of this, you will just return 1 here.

```
get numComponents() : number { return 1 }
```